docs/submitting_block_process.md

```diff
@@ -0,0 +1,149 @@
+
+- `l2block.transactions`
+  This is L2 transations original data.
+
+  **The challenge mechanism in Optimistic Rollup ensures that l2block.transactions are correct.**
+
+- `l2block.withdrawals`
+  This is L2 withdrawals original data.
+
+  _The State Validator_ script must ensure there exists withdrawal cells in _submitting_block_transaction_'s outputs match the L2 withdrawals.
+
+- `l2block.raw.number`
+  Assert that `l2block.raw.number` must be equal to [`prev_global_state.block.count`](https://github.com/nervosnetwork/godwoken/blob/6148a733fcf7b41ce25aee3a44e8f2ae6158390d/crates/types/schemas/godwoken.mol#L29).
+
+- `l2block.raw.block_producer`
+  Skip due to out of topic of this article.
+
+- `l2block.raw.parent_block_hash`
+  Assert that `l2block.raw.parent_block_hash` must be equal [`prev_global_state.tip_block_hash`](https://github.com/nervosnetwork/godwoken/blob/6148a733fcf7b41ce25aee3a44e8f2ae6158390d/crates/types/schemas/godwoken.mol#L31).
+
+- `l2block.raw.stake_cell_owner_lock_hash`
+  Skip due to out of topic of this article.
+
+- `l2block.raw.timestamp`
+  Assert that `l2block.raw.timestamp` must be greater than [`prev_global_state.tip_block_timestamp`](https://github.com/nervosnetwork/godwoken/blob/6148a733fcf7b41ce25aee3a44e8f2ae6158390d/crates/types/schemas/godwoken.mol#L32), in order to enforce the increase of block timestamps; and less than `_submitting_block_transaction_.since`, in order to prevent the block timestamp from exceeding actual time.
+
+- `l2block.raw.prev_account`
+  Assert that `l2block.raw.prev_account` must be equal to [`prev_global_state.account`](https://github.com/nervosnetwork/godwoken/blob/6148a733fcf7b41ce25aee3a44e8f2ae6158390d/crates/types/schemas/godwoken.mol#L28).
+
+- `l2block.raw.post_account`
+  `l2block.raw.post_account` represents the global accounts tree, also called the World State. As defined in the blockchain protocol, the previous World State will transit to the new World State by applying deposits/withdrawals/L2Transactions included in the present _l2block._
+
+  _State Validator_ is able to apply the deposits and withdrawals(read more from chapter "Apply Deposits and Withdrawals"). However, as an Optimistic Rollup framework, Godwoken does not execute l2transactions on L1. This means the L1 scripts are not able to actually verify the _World State_ by directly applying l2transactions. Optimistic Rollup's challenge mechanism takes the responsibility to enforce the correctness of `L2Block.raw.post_account`.
+
+- `l2block.raw.state_checkpoint_list`
+  `l2block.raw.state_checkpoint_list` contains a list of checkpoints that represents the _World State_ after applying every transactions. Therefore, its length must be equal to `l2block.transactions`.
+
+   Besides, Optimistic Rollup's challenge mechanism takes the responsibility to enforce the correctness of `l2block.raw.state_checkpoint_list`.
+
+- `l2block.raw.submit_withdrawals`
+  ❓
+
+- `l2block.raw.submit_transactions`
+  ❓
```

Comment:
A commitment to transactions included in the block. It contains transactions count and transactions merkle root.